include/circt/Dialect/Ibis/IbisOps.td

```diff
@@ -160,13 +167,48 @@ def CallOp : IbisOp<"call", [CallOpInterface]> {
   }];
 }
 
+def GetParentOp : IbisOp<"get_parent"> {
+  let summary = "Ibis get parent";
+  let description = [{
+    Given an Ibis class reference, returns the parent of said class. The
+    returned `ibis.classref` will be an opaque type.
+  }];
+
+  let arguments = (ins AnyClassRefType:$src);
+  let results = (outs OpaqueClassRefType:$parent);
+
+  let assemblyFormat = [{
+    $src `:` qualified(type($src)) attr-dict
+  }];
+}
+
+def GetChildOp : IbisOp<"get_child"> {
+  let summary = "Ibis get child";
+  let description = [{
+    Given an Ibis class reference, returns a child of said class. The child
+    is specified by the symbol name of the instance of that child in the referenced
+    class. In case of an opaque reference to the containing class, no type
+    validation is performed, and it is up to lowering passes to determine whether
+    the child of the given classref actually exists.
+    The returned classref, however, is typed, given that one must specify the
+    symbol of the class which the child is an instance of.
+  }];
+
+  let arguments = (ins
+    AnyClassRefType:$src,
+    FlatSymbolRefAttr:$childName,
+    FlatSymbolRefAttr:$childSymbol);
+  let results = (outs ClassRefType:$child);
+  let assemblyFormat = [{
+    $childName `:` $childSymbol `in` $src `:` qualified(type($src)) attr-dict custom<ClassRefFromName>(type($child), ref($childSymbol))
+  }];
+}
 
 // ===---------------------------------------------------------------------===//
 // Low-level Ibis operations
 // ===---------------------------------------------------------------------===//
 
 def ContainerOp : IbisOp<"container", [
-  IsolatedFromAbove,
```

Comment:
I'm not a big fan of using "IsolatedFromAbove"... mostly because I'm lazy but also because I think it's too restrictive in many cases. Here, I think it makes sense to drop it since at the language level, methods can implicitly access instance variables.